packages/cli/src/runners/task-runner-process.ts

```diff
@@ -63,12 +87,41 @@ export class TaskRunnerProcess {
 		this.isShuttingDown = true;
 
 		// TODO: Timeout & force kill
-		this.process.kill();
+		if (this.globalConfig.taskRunners.useLauncher) {
+			await this.killLauncher();
+		} else {
+			this.killNode();
+		}
 		await this.runPromise;
 
 		this.isShuttingDown = false;
 	}
 
+	killNode() {
+		if (!this.process) {
+			return;
+		}
+		this.process.kill();
+	}
+
+	async killLauncher() {
+		if (!this.process?.pid) {
+			return;
+		}
+
+		const process = spawn(this.globalConfig.taskRunners.launcherPath, [
+			'kill',
+			this.globalConfig.taskRunners.launcherRunner,
+			this.process.pid.toString(),
+		]);
+
+		await new Promise<void>((resolve) => {
+			process.on('exit', () => {
+				resolve();
+			});
+		});
```

Comment:
Could the kill process fail for some reason? Should we log an error in that case?